distributed/scheduler.py

```diff
@@ -4872,6 +4872,13 @@ async def update_graph(
                 global_annotations=annotations or {},
                 validate=self.validate,
             )
+
+            materialization_done = time()
+            logger.debug("Materialization done. Got %i tasks.", len(dsk))
+            dependents = reverse_dict(dependencies)
+            dsk = resolve_aliases(dsk, keys, dependents)
+            dependencies = dict(DependenciesMapping(dsk))
+            logger.debug("Removing aliases. %i left", len(dsk))
```

Comment:
This is new... it removes all those 

`"key": "other-key"`

references that we currently schedule as real tasks. Particularly fusion adds these kinds of redirects. I should probably factor this out to a dedicated PR. The impact can be quite substantial in graph size reduction.